etc/elasticsearch.yml

```diff
@@ -0,0 +1,130 @@
+# etc/elasticsearch.yml
+AWSTemplateFormatVersion: "2010-09-09"
+Description: >
+  Creates an Amazon Elasticsearch Service (Amazon ES) domain, with a VPC
+  endpoint, automated snapshots, and an optional EBS volume.
+Parameters:
+  ProjectTag:
+    Type: String
+  EnvironmentTag:
+    Type: String
+    AllowedValues:
+      - Staging
+      - Production
+  EbsEnabled:
+    Type: String
+    AllowedValues:
+      - true
+      - false
+  EbsVolumeType:
+    Type: String
+    AllowedValues:
+      - standard
+      - gp2
+  EbsVolumeSize:
+    Type: Number
+  ElasticsearchVersion:
+    Type: String
+    AllowedValues:
+      - 6.4
+  ElasticsearchInstanceCount:
+    Type: Number
+  ElasticsearchInstanceType:
+    Type: String
+    AllowedValues:
+      - t2.small.elasticsearch
+  ElasticsearchZoneAwarenessEnabled:
+    Type: String
+    AllowedValues:
+      - true
+      - false
+  VpsSecurityGroupIds:
+    Type: List<AWS::EC2::SecurityGroup::Id>
+  VpcSubnetIds:
+    Type: List<AWS::EC2::Subnet::Id>
+  AutomatedSnapshotStartHour:
+    Type: Number
+Metadata:
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: Elasticsearch
+        Parameters:
+          - ElasticsearchVersion
+          - ElasticsearchInstanceCount
+          - ElasticsearchInstanceType
+          - ElasticsearchZoneAwarenessEnabled
+          - AutomatedSnapshotStartHour
+      - Label:
+          default: EBS
+        Parameters:
+          - EbsEnabled
+          - EbsVolumeSize
+          - EbsVolumeType
+      - Label:
+          default: Resource Tagging
+        Parameters:
+          - ProjectTag
+          - EnvironmentTag
+    ParameterLabels:
+      EbsEnabled:
+        default: Enabled
+      EbsVolumeType:
+        default: Volume Type
+      EbsVolumeSize:
+        default: Volume Size
+      ProjectTag:
+        default: Project Tag
+      ElasticsearchVersion:
+        default: Version
+      ElasticsearchInstanceCount:
+        default: Instance Count
+      ElasticsearchInstanceType:
+        default: Data Node Instance Type
+      ElasticsearchZoneAwarenessEnabled:
+        default: Multi-AZ
+      AutomatedSnapshotStartHour:
+        default: Automated Snapshot Hour
+Resources:
+  ElasticsearchDomain:
+    Type: AWS::Elasticsearch::Domain
+    Properties:
+      AccessPolicies:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              AWS: "*"
+            Action: "es:*"
+            Resource: !Sub "arn:aws:es:${AWS::Region}:${AWS::AccountId}:domain/*"
+  # #     AdvancedOptions:
+  # #       String: String
+      EBSOptions:
+        EBSEnabled: !Ref EbsEnabled
```

Comment:
Is this really optional though?  In the web console, I can't even select "Instance Storage" - it's greyed out.

Reply:
The API docs make it seem very optional. For now I kept the parameter, but forced it to always be `true`